.github/workflows/docs.yml

```diff
@@ -0,0 +1,32 @@
+name: Build and Deploy Docs
+
+on:
+  push:
+    branches: [main]
+
+permissions:
+  contents: write
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Build developer documentation
+        run: cargo doc --workspace --all-features --no-deps
+
+      - name: Remove leftover lock file
+        run: rm -f target/doc/.lock
```

Comment:
Why is this necessary? It seems like a workaround for some specific problem, and that problem should be documented here as a comment.